core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1073,13 +1088,29 @@ void cleanupExpiredRemoteLogSegments() throws RemoteStorageException, ExecutionE
                         .iterator();
                 while (epochsToClean.hasNext()) {
                     int epoch = epochsToClean.next();
+                    List<RemoteLogSegmentMetadata> listOfSegmentsToBeCleaned = new ArrayList<>();
+                    sizeOfDeletableSegmentsBytes = 0L;
```

Comment:
I don't think we should reset these to values. The `cleanupExpiredRemoteLogSegments` process is not completed, yet, we should add them together. 

Suppose:
phase 1: time or size or log start breached, we have total 10 segments, 100 GB to be deleted.
RemoteDeleteBytesLag: 100GB, 
RemoteDeleteSegmentsLag: 10

phase 2: delete all these segments 1 by 1, but unfortunately, 1 segment and 10 GB size cannot be deleted
RemoteDeleteBytesLag: 10GB, 
RemoteDeleteSegmentsLag: 1

phase 3: check if epoch is unreferenced, we have 2 segments, 20 GB to be deleted. In this case:
RemoteDeleteBytesLag: _should be 10GB + 20GB_ 
RemoteDeleteSegmentsLag: _should be 1 + 2_

Does that make sense?

Reply:
Yup, to paraphrase - we could have leftover undeleted segments and we need to carry them forward in the calculations. I have amended this in subsequent commits!